<h3 style="margin:0px">Class: org.apache.commons.jexl3.junit.AsserterTest (2 methods) </h3><br><pre style="padding:0"><code class="html"><div id="tags"><button id="13"style="border-width:0;padding:0;background:#282B2E"  onclick="filter('13')" data-toggle="tooltip" title="Verifies (un)successful execution of the test case by reporting explicitly a failure"><kbd id="tag-13"class="label-info"style="display: inline-block;font-size:7pt" >UtilityVerifier&nbsp;(2)</kbd></button>&nbsp;</div></code></pre><center><button onclick="showBlocks()" style="margin:6px">Update filter <span class="glyphicon glyphicon-refresh" aria-hidden="true"></span> </button></center><br>
<pre class="type-13 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies (un)successful execution of the test case by reporting explicitly a failure">UtilityVerifier</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies (un)successful execution of the test case by reporting explicitly a failure
"></span><br>
@Test public void testVariable() throws Exception {
  Asserter asserter=new Asserter(JEXL);
  asserter.setSilent(true);
  asserter.setVariable("foo",new Foo());
  asserter.setVariable("person","James");
  asserter.assertExpression("person","James");
  asserter.assertExpression("size(person)",new Integer(5));
  asserter.assertExpression("foo.getCount()",new Integer(5));
  asserter.assertExpression("foo.count",new Integer(5));
  try {
    asserter.assertExpression("bar.count",new Integer(5));
    Assert.fail("This method should have thrown an assertion exception");
  }
 catch (  AssertionFailedError e) {
  }
}

</code></pre>

<pre class="type-13 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies (un)successful execution of the test case by reporting explicitly a failure">UtilityVerifier</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies (un)successful execution of the test case by reporting explicitly a failure
"></span><br>
@Test public void testThis() throws Exception {
  Asserter asserter=new Asserter(JEXL);
  asserter.setVariable("this",new Foo());
  asserter.assertExpression("this.get('abc')","Repeat : abc");
  try {
    asserter.assertExpression("this.count","Wrong Value");
    Assert.fail("This method should have thrown an assertion exception");
  }
 catch (  AssertionFailedError e) {
  }
}

</code></pre>

<script>$(document).ready(function() {
  $('pre code').each(function(i, block) {
    hljs.highlightBlock(block);
  });
});</script>
